xocto/storage/storage.py

```diff
@@ -126,6 +126,64 @@ class ReadableBinaryFile(Protocol):
     def read(self, size: int = ...) -> bytes: ...
 
 
+def make_boto_config(
+    *,
+    connect_timeout: int | None = None,
+    read_timeout: int | None = None,
+    total_max_attempts: int | None = None,
+) -> botocore.config.Config:
+    """
+    Used to make a botocore.config.Config for use with the S3FileStore class.
+
+    By default that class will use `make_boto_config_from_django_settings` which
+    will populate the inputs to this function using django settings
+    """
+    config: dict[str, object] = {}
+
+    if connect_timeout is not None:
+        config["connect_timeout"] = connect_timeout
+
+    if read_timeout is not None:
+        config["read_timeout"] = read_timeout
+
+    if total_max_attempts is not None:
+        config["retries"] = {"total_max_attempts": total_max_attempts}
+
+    return botocore.config.Config(**config)
+
+
+def make_boto_config_from_django_settings(
+    *,
+    connect_timeout_key: str = "BOTO_S3_CONNECT_TIMEOUT",
+    read_timeout_key: str = "BOTO_S3_READ_TIMEOUT",
+    total_max_attempts_key: str = "BOTO_S3_TOTAL_MAX_ATTEMPTS",
+) -> botocore.config.Config:
+    """
+    Used to make a botocore.config.Config for use with the S3FileStore class.
+
+    This will find django settings using the specified keys and ensure if those have values,
+    that those values are integers, before passing into `make_boto_config` to make a boto config
+    """
+
+    def _get_int(key: str) -> int | None:
+        val = getattr(settings, key, None)
+        if val is None or isinstance(val, int):
+            return val
+
+        if isinstance(val, str) and val.isdigit():
+            return int(val)
+
+        raise ValueError(
```

Comment:
Should this be `TypeError`?

> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

https://docs.python.org/3/library/exceptions.html#TypeError